section_faiss_vector_index.go

```diff
@@ -510,32 +510,41 @@ func (vo *vectorIndexOpaque) process(field index.VectorField, fieldID uint16, do
 	dim := field.Dims()
 	metric := field.Similarity()
 
-	if vec != nil {
+	// Embedder is supposed to make sure len(vec) is a multiple of dim.
+	// Not double checking it here to avoid the overhead.
+	numSubVecs := len(vec) / int(dim)
+	for i := 0; i < numSubVecs; i++ {
+		subVec := vec[i*int(dim) : (i+1)*int(dim)]
+
+		if subVec == nil {
+			continue
+		}
+
 		// NOTE: currently, indexing only unique vectors.
-		vecHash := hashCode(vec)
-		if _, ok := vo.vecIDMap[vecHash]; !ok {
-			vo.vecIDMap[vecHash] = vecInfo{
+		subVecHash := hashCode(subVec)
+		if _, ok := vo.vecIDMap[subVecHash]; !ok {
+			vo.vecIDMap[subVecHash] = vecInfo{
 				docIDs: roaring.NewBitmap(),
 			}
 		}
 		// add the docID to the bitmap
-		vo.vecIDMap[vecHash].docIDs.Add(docNum)
+		vo.vecIDMap[subVecHash].docIDs.Add(docNum)
 
 		// tracking the unique vectors for every field which will be used later
 		// to construct the vector index.
 		if _, ok := vo.vecFieldMap[fieldID]; !ok {
 			vo.vecFieldMap[fieldID] = indexContent{
```

Comment:
Optimization:

vecIDMap can be restructured to store only slice of hashValues.
Instead of a map of <hashValue, vecInfo>

Reply:
@moshaad7 Both this and the earlier optimization you cite sound good to accommodate within these PRs.

Reply:
Sorry I wanted to say something else, but typed other thing.

We have two maps and redundant content in them

```

vecIDMap map[int64]vecInfo
vecFieldMap map[uint16]indexContent

type vecInfo struct {
	vec    []float32
	docIDs *roaring.Bitmap
}

type indexContent struct {
	vecs   map[int64]vecInfo
	dim    uint16
	metric string
}
```

You can see that vecInfo is redundant in both indexContent and vecIDMap. 

So, indexContent should only store vector hashes, and if it want to know the vecInfo, it should consult vecIDMap

Will continue this conversation in new PR